Deferred/Deferred.swift

```diff
@@ -62,21 +70,38 @@ public struct Deferred<Value> {
         }
     }
 
-    // Initialize an unfilled Deferred
+    /// Initialize an unfilled Deferred.
     public init() {
         self.init(nil)
     }
 
-    // Initialize a filled Deferred with the given value
+    /// Initialize a filled Deferred with the given value.
     public init(value: Value) {
         self.init(value)
     }
 
-    // Check whether or not the receiver is filled
+    /// Check whether or not the receiver is filled.
     public var isFilled: Bool {
         return dispatch_block_wait(onFilled, DISPATCH_TIME_NOW) == 0
     }
 
+    /// Determines the deferred value with a given result.
+    ///
+    /// Filling a deferred value should usually be attempted only once, and by
+    /// default filling will trap upon improper usage.
+    ///
+    /// * In playgrounds and unoptimized builds (the default for a "Debug"
+    ///   configuration), program execution will be stopped at the caller in
+    ///   a debuggable state.
+    /// * In -O builds (the default for a "Release" configuration), program
+    ///   execution will stop.
+    /// * In -Ounchecked builds, the programming error is assumed to not exist.
+    ///
+    /// If your deferred requires multiple potential fillers to race, you may
+    /// disable the precondition.
+    ///
+    /// :param: value The resolved value of the deferred.
+    /// :param: assertIfFilled If `false`, race checking is disabled.
     public func fill(value: Value, assertIfFilled: Bool = true, file: StaticString = __FILE__, line: UWord = __LINE__) {
```

Comment:
The parameter `assertIfFilled` alone leaves me confused as to whether it's "assert if this `fill` call fills it" or "assert if it's already filled". It's clear from the comment and some thinking that it's the latter, but is there some alternative parameter name we could choose that would make it immediately clear at call sites, too?